# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xpdf/xpdf-3.02pl1.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

*** xpdf-3.02.orig/xpdf/Stream.cc	Tue Feb 27 14:05:52 2007
--- xpdf-3.02/xpdf/Stream.cc	Thu Jul 26 14:44:43 2007
***************
*** 410,424 ****
    ok = gFalse;
  
    nVals = width * nComps;
-   if (width <= 0 || nComps <= 0 || nBits <= 0 ||
-       nComps >= INT_MAX / nBits ||
-       width >= INT_MAX / nComps / nBits ||
-       nVals * nBits + 7 < 0) {
-     return;
-   }
    pixBytes = (nComps * nBits + 7) >> 3;
    rowBytes = ((nVals * nBits + 7) >> 3) + pixBytes;
!   if (rowBytes <= 0) {
      return;
    }
    predLine = (Guchar *)gmalloc(rowBytes);
--- 410,422 ----
    ok = gFalse;
  
    nVals = width * nComps;
    pixBytes = (nComps * nBits + 7) >> 3;
    rowBytes = ((nVals * nBits + 7) >> 3) + pixBytes;
!   if (width <= 0 || nComps <= 0 || nBits <= 0 ||
!       nComps > gfxColorMaxComps ||
!       nBits > 16 ||
!       width >= INT_MAX / nComps ||      // check for overflow in nVals 
!       nVals >= (INT_MAX - 7) / nBits) { // check for overflow in rowBytes
      return;
    }
    predLine = (Guchar *)gmalloc(rowBytes);
